benchmark/benchmark_utils.py

```diff
@@ -68,26 +71,28 @@ def benchmark(self, n):
         """
 
 
-def create_qnode(qfunc, device, mutable=True, interface="autograd"):
+def create_qnode(qfunc, device, mutable=True, interface="autograd", qnode_type="QNode"):
     """Utility function for creating a quantum node.
 
     Takes care of the backwards compatibility of the benchmarks.
 
-    Implicitly uses the parameter-shift method for computing Jacobians.
+    By default, uses the parameter-shift method for computing Jacobians.
 
     Args:
         qfunc (Callable): quantum function defining a circuit
         device (~pennylane.Device): device for executing the circuit
         mutable (bool): whether the QNode should mutable
         interface (str, None): interface used for classical backpropagation,
             in ('autograd', 'torch', 'tf', None)
+        qnode_type (str): name of the specific QNode subclass to use
 
     Returns:
         BaseQNode: constructed QNode
     """
     try:
-        qnode = qml.qnodes.QNode(
-            qfunc, device, mutable=mutable, diff_method="parameter-shift", interface=interface,
+        qnode_type = getattr(qml.qnodes, qnode_type)
+        qnode = qnode_type(
+            qfunc, device, mutable=mutable, interface=interface,
```

Comment:
Small comment --- previously, an attribute error would be raised if `qml.qnodes.QNode` did not exist, which would indicate to the benchmarking script that an older version of PL was being used before the QNode refactor.

However, the change of logic here means that an attribute error could be raised for two different reasons:

1. Version of PL is after the QNode refactor, but the user specified `qnode_type` doesn't exist/user had a typo.

2. Version of PL is before the QNode refactor.

This could be fixed by separating checking if `qml.qnodes` exists (`if hasattr(qml, "qnodes")`) and then having a separate `try-except Attribute error` branch. Or maybe by using the form `getattr(qml.qnodes, qnode_type, [default value if attribute error])`.

Reply:
Since the benchmarking tool is not officially part of the codebase, it's not even unit testesd.

In this case, I'd rather keep the tool as-is for the moment, and just recognize that it has some rough edges, rather than worrying too much about bringing up the standard.